types/mocha/index.d.ts

```diff
@@ -389,6 +389,8 @@ declare namespace Mocha {
          * - _Only available when invoked via the mocha CLI._
          */
         (fn: Func): void;
+        // eslint-disable-next-line @definitelytyped/no-unnecessary-generics
+        <T extends BaseContext>(fn: Func<T>): void;
```

Comment:
Would `<T extends BaseContext = Context>(fn: Func<T>): void;` allow you to remove the non-generic version?

Reply:
No, because then every time I invoke the function, I would have to call it with an explicit subtype, or the compiler would assume Context.  This way, the compiler infers the type.  It's ugly, I know.